### PR TITLE
port the atbash-cipher exercism to perl6

### PR DESCRIPTION
**Pull request by yanick**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
       "slug": "bob",
       "topics": []
     },
+    { "difficulty" : 1, "slug" : "atbash-cipher", "topics" : [] },
```
Comment (mienaikage) [inline config.json]: Can we have this object on multiple lines like the others?

File: exercises/atbash-cipher/cipher.t [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+subtest 'decode' => {
+    my @cases = |%cases.<decode>.<cases>;
+
+    plan +@cases;
+
+    is decode( .<phrase> ), .<expected>, .<description>
+        for @cases;
+};
+
+done-testing;
```
Comment (mienaikage) [inline exercises/atbash-cipher/cipher.t]: Don't need done-testing here :)

Comment (yanick) [inline exercises/atbash-cipher/cipher.t]: It's not mandatory, but it doesn't hurt to be explicit.

Comment (mienaikage) [inline exercises/atbash-cipher/cipher.t]: `done-testing` is usually for when you can't/don't have a `plan`, I don't think it's a good idea to keep them both together. The testing docs recommend that it be removed so I don't think we should keep it in without a good reason for doing so.

Comment (yanick) [inline exercises/atbash-cipher/cipher.t]: > I don't think it's a good idea to keep them both together.

What is the disadvantage?

Comment (mienaikage) [inline exercises/atbash-cipher/cipher.t]: 1. Having them both together results in `done-testing` doing nothing. The disadvantage of including code which does nothing is potential confusion for people reading or modifying said code. This is especially true for people doing exercises on the site who may be new to Perl 6. Furthermore, if someone modifies the code in future and changes the `plan` line, they may miss this line and commit changes without realising it has taken effect.

2. Re: being explicit, why have a function when you could have a comment which would definitely do nothing? Following that, would you see there being any advantage to leaving a comment at the end of the test?

Comment (yanick) [inline exercises/atbash-cipher/cipher.t]: `done-testing` here is akin to `return` as the last statement of a function. Perl would return the value yielded by the last statement with or without the presence of the keyword `return`, but its presence is a visual cue to the reader that the function is expected to return/end at this point. It is also more than what a comment would provide as it also acts as a sanity check that will dot and cross all the TAP Is and Ts.

This being said, while I don't feel like changing the code for something I feel is spurious, I won't mind if you do so alter the branch yourself. Go wild. :-)

File: exercises/atbash-cipher/cipher.t [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env perl6
+
+use Test;
+use JSON::Tiny;
+
+use lib ( my $dir = IO::Path.new($?FILE).parent ).path;
+
+my $module_name = %*ENV<EXERCISM>.so ?? 'Example' !! 'Allergies';
```
Comment (4d47) [inline exercises/atbash-cipher/cipher.t]: This looks like copy/paste error, should probably try to load 'Atbash' instead

Comment (yanick) [inline exercises/atbash-cipher/cipher.t]: Gah! You're right!

**Pull request merged**